### PR TITLE
fix(frontend): security hardening for Electron, UI inputs, and store logic

### LINKED ISSUE #1330: [BUG] Frontend/Electron security: XSS in file parsers, unsigned updates, arbitrary file exec, weak RNG, secret leak
### Summary

Nine security and correctness issues found across the Electron main process and frontend codebase during a security audit.

#### 1. HIGH - Disabled Code Signature Verification (electron/main/update.ts)
`verifyUpdateCodeSignature` is set to `false`, allowing unsigned or tampered updates to be installed without verification.

**Impact**: Malicious update injection via unsigned packages.

#### 2. HIGH - XSS in File Parsers (electron/main/fileReader.ts)
xlsx, csv, and pptx cell content is rendered as raw HTML without escaping. Malicious spreadsheet files can execute arbitrary JavaScript in the Electron renderer.

**Impact**: Code execution via crafted spreadsheet/document files.

#### 3. HIGH - Event Listener Accumulation (electron/main/index.ts)
`startDownload` adds new listeners on each call without removing previous ones. Over time this causes memory leaks and duplicate event handling.

**Impact**: Memory leak and duplicate event processing.

#### 4. HIGH - Arbitrary File Execution (electron/main/index.ts)
`shell.openPath()` directly executes files. If the user opens a malicious file, it runs with the user's privileges instead of just revealing it in the file manager.

**Impact**: Arbitrary code execution via file open action.

#### 5. HIGH - OAuth Token Leak in Logs (electron/main/index.ts)
Authorization codes are logged in plaintext, exposing them in log files and console output.

**Impact**: OAuth token exposure via log files.

#### 6. MEDIUM - Division by Zero in Task Progress (src/store/chatStore.ts)
`taskProgress` calculation crashes when `taskRunning` is empty, dividing by zero.

**Impact**: UI crash when viewing empty task progress.

#### 7. HIGH - XSS via dangerouslySetInnerHTML (src/components/ui/input.tsx, textarea.tsx)
User-influenced HTML is rendered via `dangerouslySetInnerHTML` without sanitization.

**Impact**: Stored XSS via form input rendering.

#### 8. HIGH - Weak RNG for ID Generation (src/lib/index.ts)
`generateUniqueId()` uses `Math.random()` which is not cryptographically secure. IDs can be predicted or collide.

**Impact**: Predictable identifiers, potential session fixation.

#### 9. MEDIUM - Secret Key Client Exposure (src/lib/index.ts)
`hasStackKeys()` references `VITE_STACK_SECRET_SERVER_KEY`, which Vite bundles into the client JavaScript. The server secret key is exposed in the browser bundle.

**Impact**: Server secret key leaked to all frontend users.

### Affected Files
- `electron/main/update.ts`
- `electron/main/fileReader.ts`
- `electron/main/index.ts` (3 bugs)
- `src/store/chatStore.ts`
- `src/components/ui/input.tsx`
- `src/components/ui/textarea.tsx`
- `src/lib/index.ts` (2 bugs)

### PR DESCRIPTION
**Pull request by eren-karakus0**
### Related Issue

Closes #1330

Split from #1299 as requested by @bytecii.

## Summary

Security hardening for the **frontend and Electron** codebase.

### Changes
- **fileReader.ts**: Sanitize file paths to prevent path traversal
- **index.ts (electron)**: Harden IPC handlers
- **update.ts**: Auto-update signature verification hardening
- **input.tsx / textarea.tsx**: Add maxLength and input sanitization
- **chatStore.ts**: Fix division-by-zero in percentage calculation
- **index.ts (lib)**: Sanitize URLs to prevent XSS via javascript: protocol

### Tests
- `fileReader.test.ts` — new, covers path traversal prevention
- `securityFixes.test.ts` — new, covers URL sanitization
- `chatStore-divisionByZero.test.ts` — new, covers division-by-zero edge case

### Files (10)
- `electron/main/fileReader.ts`
- `electron/main/index.ts`
- `electron/main/update.ts`
- `src/components/ui/input.tsx`
- `src/components/ui/textarea.tsx`
- `src/lib/index.ts`
- `src/store/chatStore.ts`
- `test/unit/electron/fileReader.test.ts`
- `test/unit/lib/securityFixes.test.ts`
- `test/unit/store/chatStore-divisionByZero.test.ts`

### CODE REVIEW

**Comment by Wendong-Fan** [comment]
thanks @eren-karakus0  for the contribution, could you link the issue this pr want to resolve? maybe @bytecii and @a7m-1st could help reviewing this

**Comment by eren-karakus0** [comment]
Linked - closes #1330. Thanks for the reminder @Wendong-Fan!

File: electron/main/index.ts [hunk 1]
```diff
@@ -862,405 +861,6 @@ function registerIpcHandlers() {
     }
   });
 
-  // ======================== skills ========================
```
Comment (bytecii) [inline electron/main/index.ts:865]: Why we remove these?

Comment (eren-karakus0) [reply electron/main/index.ts:865]: This was an accidental deletion. The skills section, unzipper import, and seedDefaultSkillsIfEmpty() call have all been restored. The only intended changes in index.ts are the 3 security fixes (OAuth token redaction, shell.showItemInFolder, log level downgrade). Fixed in latest push.

**Comment by eren-karakus0** [comment]
Hey @bytecii, just a friendly ping - all review feedback has been addressed and CI is green (7/7). Could you take another look when you get a chance? Thanks!

**Comment by eren-karakus0** [comment]
Friendly ping - just checking if there's anything I should update or adjust in this PR. Happy to address any feedback\!
